pkg/component/monitoring/prometheusoperator/deployment.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusoperator
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/utils"
+)
+
+func (p *prometheusOperator) deployment() *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "prometheus-operator",
+			Namespace: p.namespace,
+			Labels:    GetLabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             pointer.Int32(1),
+			RevisionHistoryLimit: pointer.Int32(2),
+			Selector:             &metav1.LabelSelector{MatchLabels: GetLabels()},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: utils.MergeStringMaps(GetLabels(), map[string]string{
+						v1beta1constants.LabelNetworkPolicyToDNS:              v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToRuntimeAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+					}),
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: serviceAccountName,
+					PriorityClassName:  p.values.PriorityClassName,
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot:   pointer.Bool(true),
+						RunAsUser:      pointer.Int64(65532),
+						SeccompProfile: &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            "prometheus-operator",
+							Image:           p.values.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args: []string{
+								fmt.Sprintf("--prometheus-config-reloader=%s", p.values.ImageConfigReloader),
```

Comment:
The deployment in `bundle.yaml` you referred to in the PR description includes the argument `--kubelet-service=kube-system/kubelet` too.
I don't really know if we need prometheus-operator to manage this service or not. I just want to ensure that it was left out by purpose and not by accident 😄

Reply:
Thanks, yes, I've seen that, but omitted it on purpose for now. I might reintroduce it when working on the migration of the respective Prometheis if needed, though. :)